sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFn.java

```diff
@@ -725,14 +731,30 @@ public Duration getAllowedTimestampSkew() {
    * href="https://s.apache.org/splittable-do-fn">splittable</a> {@link DoFn} into multiple parts to
    * be processed in parallel.
    *
-   * <p>Signature: {@code List<RestrictionT> splitRestriction( InputT element, RestrictionT
-   * restriction);}
+   * <p>The signature of this method must satisfy the following constraints:
+   *
+   * <ul>
+   *   <li>One of its parameters must be the {@code InputT} element.
+   *   <li>One of its parameters must be the restriction.
+   *   <li>One of its input parameters must be of type {@link Backlog}. Splitting the restriction
+   *       should attempt to take the backlog information into account. If the backlog is known,
+   *       each split should return a restriction with an approximate amount of work bounded by the
+   *       backlog. In the case of an unbounded restriction, at most one of the splits can represent
+   *       the unbounded portion of work. If the backlog that is specified is unknown, it is up to
+   *       the SDK to choose a number of splits of approximately equally sized portions with
+   *       potentially one of those splits representing the unbounded portion of work.
+   *   <li>One of its parameters must be the output receiver for restrictions.
+   * </ul>
+   *
+   * <p>Signature: {@code splitRestriction(InputT element, RestrictionT restriction, Backlog
+   * backlog, OutputReceiver<RestrictionT> receiver);}
    *
    * <p>Optional: if this method is omitted, the restriction will not be split (equivalent to
-   * defining the method and returning {@code Collections.singletonList(restriction)}).
+   * defining the method and outputting {@code Collections.singletonList(restriction)}).
+   *
+   * <p>TODO: Make the InputT parameter optional.
```

Comment:
This refers to the implementation in the invoker part, is it? Maybe worth doing it as part of this.